File: rust_package/brainflow/examples/ica.rs
use ndarray::Array2;
use std::{thread, time::Duration};

use brainflow::{
    board_shim, brainflow_input_params::BrainFlowInputParamsBuilder, data_filter, BoardIds, BrainFlowPresets,
};
use ndarray::s;

fn main() {
    brainflow::board_shim::enable_dev_board_logger().unwrap();

    let board_id = BoardIds::SyntheticBoard;
    let eeg_channels = board_shim::get_eeg_channels(board_id, BrainFlowPresets::DefaultPreset).unwrap();
    println!("{:?}", eeg_channels);

    let params = BrainFlowInputParamsBuilder::default().build();
    let board = board_shim::BoardShim::new(board_id, params).unwrap();

    board.prepare_session().unwrap();
    board.start_stream(45000, "").unwrap();
    thread::sleep(Duration::from_secs(10));
    board.stop_stream().unwrap();
    let mut data = board.get_board_data(Some(500), BrainFlowPresets::DefaultPreset).unwrap();
    board.release_session().unwrap();

    let my_channel = eeg_channels[4];
    let mut my_data = data.slice_mut(s![my_channel, ..]);
    let data_my_channel = my_data.as_slice_mut().unwrap();
    let ica_data = Array2::from_shape_vec((5, 100), data_my_channel.to_vec()).unwrap();
    let _ica = data_filter::perform_ica(ica_data, 2).unwrap();
}

File: matlab_package/brainflow/examples/ICA.m
params = BrainFlowInputParams();
board_shim = BoardShim(int32(BoardIds.SYNTHETIC_BOARD), params);
preset = int32(BrainFlowPresets.DEFAULT_PRESET);
board_shim.prepare_session();
board_shim.start_stream(45000, '');
pause(10);
board_shim.stop_stream();
data = board_shim.get_board_data(500, preset);
board_shim.release_session();

eeg_channels = BoardShim.get_eeg_channels(int32(BoardIds.SYNTHETIC_BOARD), preset);
selected_channel = eeg_channels(4);
original_data = data(selected_channel, :);
original_data = transpose(reshape(original_data, [100, 5]));
[w,k,a,s] = DataFilter.perform_ica(original_data, 2);